js/playlist.js

```diff
@@ -21,9 +23,12 @@ function(ytData,ytIframe,template,$) {
         } else {
             if(JSON.parse(localStorage.getItem('new_playlist_saved'))) {
                 playlist = JSON.parse(localStorage.getItem('playlist'));
-                var rptst = localStorage.getItem('repeat');
-
-                total_duration = JSON.parse(localStorage.getItem('total_duration'));
+                if (curr_id==undefined) curr_id=playlist.length-1;
+                console.log(curr_id);
+                var rptst = playlist[curr_id]['repeat'];
```

Comment:
Here you access `playlist[curr_id]` as a dict, whereas later in the loop, you are accessing it as a list. Have you tested this code?